lib/widgets/errors/medito_error_widget.dart

```diff
@@ -21,26 +21,42 @@ class MeditoErrorWidget extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) {
-    var isInvalidToken = message == StringConstants.invalidToken;
+    var splittedMessage = message.split(': ');
+    var _showCheckDownloadText = showCheckDownloadText;
 
+    if (splittedMessage.length > 1) {
+      var statusCode = int.parse(splittedMessage[1]);
```

Comment:
This could throw an exception. I would split all this logic out into another method and also handle the exception.